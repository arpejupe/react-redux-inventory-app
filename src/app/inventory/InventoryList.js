import React from 'react'
import PropTypes from 'prop-types';

import CircularProgress from '@material-ui/core/CircularProgress';
import Container from '@material-ui/core/Container';

import { Link } from 'react-router-dom';

import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import { Typography } from '@material-ui/core';
import Whatshot from '@material-ui/icons/Whatshot';

import { withStyles } from '@material-ui/core/styles';

const styles = {
  inventoryList: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
    marginTop: '1.5em'
  },
  inventoryBox: {
    backgroundColor: 'white',
    textAlign: 'center'
  },
  inventoryIcon: {
    marginBottom: '0.05em',
    fontSize: '4.5rem',
  },
  inventoryHeader: {
    marginLeft: '1em',
    fontSize: '15px'
  },
  progress: {
    marginTop: '40%',
    marginLeft: '50%'
  },
  inventoryLink: {
    textDecoration: 'none',
  }
};

class InventoryList extends React.Component {
  
  componentDidMount() {
    this.props.actions.fetchInventory();
  }

  render() {
    if (this.props.loading) {
      return (
        <Container className={this.props.classes.inventoryList}>
          <CircularProgress 
            className={this.props.classes.progress}
          />
        </Container> 
      );
    } else {
      return (
        <Container className={this.props.classes.inventoryList}>
          <Typography className={this.props.classes.inventoryHeader} 
                      variant="overline" 
                      display="block" 
                      gutterBottom>
                        My Inventory
          </Typography>
          <Grid container>
            {this.props.inventory.map(inv => (
              <Link key={inv.id} to={`inventory/${inv.id}`} className={this.props.classes.inventoryLink} variant="inherit" color="inherit" underline="none">
                <Box className={this.props.classes.inventoryBox} boxShadow={1} key={inv.id} p={3} m={2}>      
                  <Whatshot className={this.props.classes.inventoryIcon} color="secondary"></Whatshot>
                  <Typography align="center" color="secondary">{inv.name}</Typography>
                </Box>
              </Link>
            ))}
          </Grid>
        </Container> 
      );
    }
  }
}

export default withStyles(styles)(InventoryList);

InventoryList.propTypes = {
  inventory: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
};
