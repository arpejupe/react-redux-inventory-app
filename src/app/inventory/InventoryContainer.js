import { bindActionCreators, } from 'redux';
import { connect } from 'react-redux'
import * as actions from './InventoryActions';
import InventoryList from './InventoryList';

const mapStateToProps = (state) => {
    return {
        loading: state.InventoryReducer.loading,
        inventory: state.InventoryReducer.inventory
    };
};
const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(actions, dispatch)
    }
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(InventoryList);