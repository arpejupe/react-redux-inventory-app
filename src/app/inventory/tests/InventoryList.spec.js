import React from "react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";

import InventoryContainer from "./../InventoryContainer";
import InventoryList from "./../InventoryList";

import Box from '@material-ui/core/Box';
import CircularProgress from '@material-ui/core/CircularProgress';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { createShallow } from '@material-ui/core/test-utils';

Enzyme.configure({ adapter: new Adapter() });

function createTestProps(props) {
    return {
      inventory: [
        {
          id: "A429",
          name: "Inventory A429",
          stock: 145,
          total: 200
        },
        {
          id: "A342",
          name: "Inventory A342",
          stock: 293,
          total: 490
        }
      ],
      loading: false,
      actions: {
        fetchInventory: jest.fn(),
      },
      ...props,
    };
}

describe('<InventoryList /> rendering', () => {

  let shallow, wrapper, minProps;

  beforeEach(() => {
    minProps = createTestProps();
    shallow = createShallow();
    wrapper = shallow(<InventoryList {...minProps} />);
  });  

  afterEach(() => {
    wrapper.unmount();
  });

  it("should match snapshot", () => {
    const state = {
      InventoryReducer: {
        loading: true,
        inventory: []
      }
    };
    const store = configureMockStore()(state);
    const component = create(
      <Provider store={store}>
        <InventoryContainer />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should fetch inventory on component mount', () => {
    const inventoryListInstance = wrapper.dive().instance();
    const fetchInventoryMock = jest.spyOn(inventoryListInstance.props.actions, 'fetchInventory');
    expect(fetchInventoryMock).toHaveBeenCalledTimes(1);
  })

  it('should render two <Box /> child components in <InventoryList />', () => {
    expect(wrapper.contains(<div className="inventoryList" />));
    expect(wrapper.dive().find(Box)).toHaveLength(2);
  });

  it('should render CircularProgress when loading is set true', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper.props().loading).toEqual(true);
    expect(wrapper.dive().find(CircularProgress)).toHaveLength(1);
  });

});
