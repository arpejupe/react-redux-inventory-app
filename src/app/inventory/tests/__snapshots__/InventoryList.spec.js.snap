// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InventoryList /> rendering should match snapshot 1`] = `
<div
  className="MuiContainer-root InventoryList-inventoryList-1 MuiContainer-maxWidthLg"
>
  <div
    className="MuiCircularProgress-root InventoryList-progress-5 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
